src/main/kotlin/com/ajax/cryptocurrency/config/SingleShotBackgroundJobBeanPostProcessor.kt

```diff
@@ -0,0 +1,42 @@
+package com.ajax.cryptocurrency.config
+
+import com.ajax.cryptocurrency.repository.CryptocurrencyRepository
+import com.ajax.cryptocurrency.threads.PriceSaver
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+@Component
+class SingleShotBackgroundJobBeanPostProcessor(
+    private val cryptocurrencyRepository: CryptocurrencyRepository,
+    @Value("\${cryptocurrency.name}")val cryptocurrencyNames: List<String>
+): BeanPostProcessor{
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any {
+        return bean
+    }
+
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any {
+        val annotation = bean::class.java.getAnnotation(SingleShotBackgroundJob::class.java)
+
+        if (annotation != null) {
+            val startDelay = annotation.startDelay
+            val maxParallelThreads = annotation.maxParallelThreads
+            val futures = mutableListOf<java.util.concurrent.Future<*>>()
+
+            Executors.newScheduledThreadPool(1).schedule({
+                val executor = Executors.newFixedThreadPool(maxParallelThreads)
+                for (cryptocurrencyName in cryptocurrencyNames) {
+                    val future = executor.submit {
+                        PriceSaver(cryptocurrencyRepository,cryptocurrencyName).run()
+                    }
```

Comment:
`PriceSaver` is runnable itself. So, this code is overcomplicated.

Two options:
1. refactor here
    ```kotlin
    executor.submit(PriceSaver(...))
    ``` 
2. Remove `Runnable` implementation in `PriceSaver`

Reply:
done